### PR TITLE
WIP: Help command aliases

### LINKED ISSUE #290: /help doesn't acknowledge aliases
/help ga
/help getarea
produce the following
image

This isn't a bug and more of a request, as it's helpful for more avenues of help to exist for users since you'd use /help when you forget something. The desired behaviour would be both '/help's above producing the same help message, rather than just the second one producing it.

### PR DESCRIPTION
**Pull request by cow-face**
Resolves #290

### CODE REVIEW

File: bin/config_sample/text/commandhelp.json [hunk 1]
```diff
@@ -1,686 +1,908 @@
...
+      "names": [
+         "lock"
+      ],
       "usage":"/lock",
       "text":"Locks the area. This command takes no arguments."
    },
    {
-      "name":"area_lock",
+      "names": [
+         "area_lock"
```
Comment (in1tiate) [inline bin/config_sample/text/commandhelp.json:224]: Merge this into `/lock` please

File: bin/config_sample/text/commandhelp.json [hunk 2]
```diff
@@ -1,686 +1,908 @@
...
+         "areakick",
+         "area_kick"
+      ],
       "usage":"/areakick <ID>",
       "text":"Kicks a client from the area, moving them back to the default area."
    },
    {
-      "name":"area_kick",
+      "names": [
+         "area_kick"
```
Comment (in1tiate) [inline bin/config_sample/text/commandhelp.json:283]: Duplicate command

File: bin/config_sample/text/commandhelp.json [hunk 3]
```diff
@@ -1,686 +1,908 @@
...
+         "notecardclear",
+         "notecard_clear"
+      ],
       "usage":"/notecardclear",
       "text":"Erases the client's notecard from the area's list of cards. A message will appear to all clients in the area indicating that a notecard has been erased. This command takes no arguments."
    },
    {
-      "name":"notecard_clear",
+      "names": [
+         "notecard_clear"
```
Comment (in1tiate) [inline bin/config_sample/text/commandhelp.json:467]: Duplicate command

File: bin/config_sample/text/commandhelp.json [hunk 4]
```diff
@@ -1,686 +1,908 @@
...
       "usage":"/judgelog",
       "text":"ends an out-of-character message with the judgelog of an area. This command takes no arguments."
    },
    {
-      "name":"allowblankposting",
+      "names": [
+         "allowblankposting",
+         "allow_blankposting"
+      ],
       "usage":"allowblankposting",
```
Comment (in1tiate) [inline bin/config_sample/text/commandhelp.json:498]:
```suggestion
      "usage":"/allowblankposting",
```

File: bin/config_sample/text/commandhelp.json [hunk 5]
```diff
@@ -1,686 +1,908 @@
...
+         "forceimmediate",
+         "force_noint_pres"
+      ],
       "usage":"/forceimmediate",
       "text":"Toggles immediate text processing in the current area. This command takes no arguments."
    },
    {
-      "name":"force_noint_pres",
+      "names": [
+         "force_noint_pres"
```
Comment (in1tiate) [inline bin/config_sample/text/commandhelp.json:616]: Merge this into `forceimmediate` please

File: bin/config_sample/text/commandhelp.json [hunk 6]
```diff
@@ -1,686 +1,908 @@
...
+      "names": [
+         "updateban"
+      ],
       "usage":"/updateban <BanID> <Field> [Duration|Reason]",
       "text":"Updates a ban in the database, changing either its reason or duration. First argument is the ban ID. Second is the field, 'reason' or 'duration'. Last argument is either the new duration or reason."
    },
    {
-      "name":"update_ban",
+      "names": [
+         "update_ban"
```
Comment (in1tiate) [inline bin/config_sample/text/commandhelp.json:778]: Merge this into `updateban` please

**Review by in1tiate — DISMISSED** [review]
Have some minor problems to address, but this looks fine otherwise

File: core/src/config_manager.cpp [hunk 7]
```diff
@@ -199,19 +199,23 @@ void ConfigManager::loadCommandHelp()
     // Akashi expects the helpfile to contain multiple entires, so it always checks for an array first.
     QJsonArray l_Json_root_array = l_help_list_json.array();
     QJsonObject l_child_obj;
+    QJsonArray l_names;
 
-    for (int i = 0; i <= l_Json_root_array.size() - 1; i++) {
+    for (int i = 0; i < l_Json_root_array.size(); i++) {
         l_child_obj = l_Json_root_array.at(i).toObject();
-        QString l_name = l_child_obj["name"].toString();
+        QJsonArray l_names = l_child_obj["names"].toArray();
```
Comment (scatterflower) [inline core/src/config_manager.cpp:206]: this variable is defined already

File: core/src/config_manager.cpp [hunk 8]
```diff
@@ -199,19 +199,23 @@ void ConfigManager::loadCommandHelp()
...
         QString l_text = l_child_obj["text"].toString();
 
-        if (!l_name.isEmpty()) {
-            help l_help_information;
-            l_help_information.usage = l_usage;
-            l_help_information.text = l_text;
+        for (int j = 0; j < l_names.size(); j++) {
+            QString l_name = l_names.at(j).toString();
+            if (!l_name.isEmpty()) {
+                help l_help_information;
```
Comment (scatterflower) [inline core/src/config_manager.cpp:213]: replace this with inline initialization i.e.

```
help l_help_information = {
     .usage = l_usage,
     .text = l_text
 };```

**Review by scatterflower — CHANGES_REQUESTED** [review]

**Comment by scatterflower** [comment]
fails tests
```
FAIL!  : tests::unittests::tst_ConfigManager::CommandInfo() Compared values are not the same
   Actual   (l_help.text)            : ""
   Expected ("A sample explanation."): A sample explanation.
   Loc: [tst_unittest_config_manager.cpp(221)]```

**Review by scatterflower — APPROVED** [review]

**Review by scatterflower — APPROVED** [review]

**Pull request merged**